src/main.js

```diff
@@ -247,6 +248,28 @@ var manifestor = function(options) {
     }
   }
 
+  // Add semantic zoom events after the first render to open
+  // the main tile source when we reach the specified zoom level on it.
+  function _semanticZoom(zoom, center) {
+    if(zoom >= _transitionZoomLevel) {
+      for(var key in _canvasObjects) {
+        if(_canvasObjects[key].containsPoint(center)) {
+          _canvasObjects[key].openMainTileSource();
+        }
+      }
+    }
+  };
+
+  viewer.addHandler('zoom', function(event) {
```

Comment:
Looks like you don't need these new event handlers now that you've added _semanticZoom to the existing event handlers below.